src/zcash/address/unified.h

```diff
@@ -7,41 +7,38 @@
 
 #include "zip32.h"
 #include "bip44.h"
-#include "zcash/Address.hpp"
 
 namespace libzcash {
 
 class ZcashdUnifiedSpendingKey;
 class ZcashdUnifiedFullViewingKey;
 
+// prototypes for the classes handling ZIP-316 encoding (in Address.hpp)
+class UnifiedAddress;
+class UnifiedFullViewingKey;
+
 class ZcashdUnifiedFullViewingKey {
 private:
-    std::optional<CExtPubKey> transparentKey;
-    std::optional<SaplingExtendedFullViewingKey> saplingKey;
+    std::optional<CChainablePubKey> transparentKey;
+    std::optional<SaplingDiversifiableFullViewingKey> saplingKey;
 
     ZcashdUnifiedFullViewingKey() {}
 
     friend class ZcashdUnifiedSpendingKey;
 public:
-    const std::optional<CExtPubKey>& GetTransparentKey() const {
+    static ZcashdUnifiedFullViewingKey FromUnifiedFullViewingKey(const UnifiedFullViewingKey& ufvk);
```

Comment:
Document that this method loses information, and should be treated as a one-way conversion.